src/ray/raylet/worker_pool.h

```diff
@@ -112,6 +112,13 @@ class WorkerPool {
   std::vector<std::shared_ptr<Worker>> GetWorkersRunningTasksForJob(
       const JobID &job_id) const;
 
+  /// Return a pointer to a worker matching the given filter function.
+  ///
+  /// \param filter Boolean function over workers.
+  /// \return A pointer to a matching worker or nullptr.
+  std::shared_ptr<Worker> FindWorker(
+      std::function<bool(std::shared_ptr<Worker>)> filter) const;
```

Comment:
Why not `const std::shared_ptr<Worker>&` here? That should avoid an increment of the refcount I think.